lib/logstash/inputs/syslog.rb

```diff
@@ -107,13 +108,32 @@ def run(output_queue)
   end # def run
 
   private
-  def udp_listener(output_queue)
-    @logger.info("Starting syslog udp listener", :address => "#{@host}:#{@port}")
+  def udp_server(output_queue)
+    udp_listener(output_queue)
+  rescue => e
+    if @shutdown_requested.false?
+      @logger.warn("syslog udp listener died", :address => "#{@host}:#{@port}", :exception => e, :backtrace => e.backtrace)
+      sleep(5)
+      retry
+    end
+  end
 
-    if @udp
-      @udp.close
+  private
+  def tcp_server(output_queue)
+    tcp_listener(output_queue)
+  rescue => e
+    if @shutdown_requested.false?
+      @logger.warn("syslog tcp listener died", :address => "#{@host}:#{@port}", :exception => e, :backtrace => e.backtrace)
+      sleep(5)
+      retry
     end
+  end
```

Comment:
Those two methods rescue all the errors from the sockets, we should be more specific with our errors handling. Some of the errors could come from the `codecs` and could be fatal error?